### PR TITLE
Allow to optionally specify custom tags for application editor, application form, and review sidebar

### PR DESCRIPTION
**Pull request by dontub**
Previously funding case types where forced to provide an AngularJS module with a custom application editor directive. This is optionally now. A default application editor is used if not specified in the funding case type. Also the application form as well as the review sidebar can be exchanged by custom ones. (See changes in documentation.)

systopia-reference: 26581

### CODE REVIEW

File: ang/crmFunding/application/applicationSidebar.directive.js [hunk 1]
```diff
@@ -16,15 +16,21 @@
 
 'use strict';
 
-fundingModule.directive('fundingApplicationSidebar', function() {
+fundingModule.directive('fundingApplicationSidebar', ['$compile', function($compile) {
   return {
     restrict: 'AE',
```
Comment (svenow) [inline ang/crmFunding/application/applicationSidebar.directive.js:21]: how to know to use `AE` (short for ApplicationEditor?)

File: ang/crmFunding/application/defaultApplicationSidebar.directive.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+fundingModule.directive('fundingDefaultApplicationSidebar', function() {
+  return {
+    restrict: 'AE',
```
Comment (svenow) [inline ang/crmFunding/application/defaultApplicationSidebar.directive.js:21]: how to know to use `AE` (short for ApplicationEditor?)

File: ang/crmFunding/application/defaultApplicationEditor.directive.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+fundingModule.directive('fundingDefaultApplicationEditor', function() {
+  return {
+    restrict: 'AE',
```
Comment (svenow) [inline ang/crmFunding/application/defaultApplicationEditor.directive.js:21]: how to know to use `AE` (short for ApplicationEditor?)

**Review by svenow — COMMENTED** [review]
just a couple of comments for clarification

File: ang/crmFunding/application/applicationEditor.directive.js [hunk 4]
```diff
@@ -30,23 +30,9 @@ fundingModule.directive('fundingApplicationEditor', ['$compile', function($compi
...
-        return name.replace(SNAKE_CASE_REGEXP, function(letter, pos) {
-          return (pos ? separator : '') + letter.toLowerCase();
-        });
-      }
-
       const unwatch = scope.$watch('fundingCaseType', function (fundingCaseType) {
         if (fundingCaseType) {
-          const directiveName = 'funding' + _4.upperFirst(fundingCaseType.name) + 'ApplicationEditor';
-          const tagName = snake_case(directiveName, '-');
+          const tagName = _4.get(fundingCaseType, 'properties.applicationEditorTagName', 'funding-default-application-editor');
```
[the remark this answers is not in the document]
Comment (dontub) [reply ang/crmFunding/application/applicationEditor.directive.js:35]: Version 4 of [lodash](https://lodash.com/). CiviCRM ships with version 3. Therefore, `_` is already reserved. Though some things aren't available in version 3. That's why funding uses version 4 with `_4`. See https://github.com/systopia/funding/blob/main/ang/lodash4.js.

File: ang/crmFunding/application/defaultApplicationSidebar.directive.js (hunk 2 above)
Comment (dontub) [reply ang/crmFunding/application/defaultApplicationSidebar.directive.js:21]: See above.

File: ang/crmFunding/application/defaultApplicationEditor.directive.js (hunk 3 above)
Comment (dontub) [reply ang/crmFunding/application/defaultApplicationEditor.directive.js:21]: See above.

**Pull request merged**